app/services/bulk_update/recommendations_uploads/validate_trainee.rb

```diff
@@ -112,17 +118,13 @@ def id_columns
 
       def trainees_by_provider_trainee_id!
```

Comment:
Could we do here:

```ruby
  def trainees_by_provider_trainee_id
       @trainees_by_provider_trainee_id ||= begin
          t = trainee_lookup[row.provider_trainee_id]
          @found_with = :provider_trainee_id if t.present?
          t
      end 
``` ?

This way I think you don't need to disable the that Cop.

Reply:
I ended up going further and rewrote the code to find the trainees with some refactoring along the way.